libV2/server/legacy.js

```diff
@@ -0,0 +1,114 @@
+const url = require('url');
+
+const config = require('../config');
+const errors = require('../errors');
+const routes = require('../../router/routes');
+const Route = require('../../router/Route');
+const Router = require('../../router/Router');
+const redisClient = require('../../utils/redisClient');
+const UtapiRequest = require('../../lib/UtapiRequest');
+const Datastore = require('../../lib/Datastore');
+
+const { LoggerContext } = require('../utils');
+
+const moduleLogger = new LoggerContext({
+    module: 'server.legacy',
+});
+
+
+/**
+ * Function to validate a URI component
+ *
+ * @param {string|object} component - path from url.parse of request.url
+ * (pathname plus query) or query from request
+ * @return {string|undefined} If `decodeURIComponent` throws an error,
```

Comment:
`_checkURIComponent` returns:
- `true` if it is an invalid string or the first invalid key if object type.
- `undefined` if the string or object is valid. 

What about having an `isValidURI` function that returns true if valid and false otherwise.

Here we are returning different types of values which can be very confusing and significantly harder to maintain.

Reply:
This class exists because the "legacy" utapi v1 server class isn't exposed. The entire function was lifted from here https://github.com/scality/utapi/blob/018a1fcdb531fda509000fdab73acd4171f851f6/lib/server.js#L48.